dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -1379,6 +1379,42 @@ void SchemaBuilder<Getter, NameMapper>::syncAllSchema()
     LOG_INFO(log, "Loaded all schemas.");
 }
 
+template <typename Getter, typename NameMapper>
+void SchemaBuilder<Getter, NameMapper>::dropAllSchema()
+{
+    LOG_INFO(log, "Dropping all schemas.");
+
+    auto & tmt_context = context.getTMTContext();
+
+    /// Drop all tables.
+    auto storage_map = tmt_context.getStorages().getAllStorage();
+    for (auto it = storage_map.begin(); it != storage_map.end(); it++)
+    {
+        auto table_info = it->second->getTableInfo();
+        if (table_info.keyspace_id != keyspace_id)
+        {
+            continue;
+        }
+        applyDropPhysicalTable(it->second->getDatabaseName(), table_info.id);
+        LOG_DEBUG(log, "Table {}.{} dropped during drop all schemas", it->second->getDatabaseName(), name_mapper.debugTableName(table_info));
+    }
```

Comment:
better to use [Range-based for loop](https://en.cppreference.com/w/cpp/language/range-for)

Reply:
@Lloyd-Pottiger This code block if copied from syncAllSchemas,  we should keep the code style same?

Reply:
no, improve it, I think.

Reply:
PTAL~